src/objects.c

```diff
@@ -959,16 +1013,28 @@ void disconnect_client(PgSocket *client, bool notify, const char *reason, ...)
 				release_server(server);
 			}
 		}
+		break;
+	case CL_ACTIVE_CANCEL:
+	case CL_WAITING_CANCEL:
+		if (client->cancel_server) {
+			PgSocket *cancel_server = client->cancel_server;
+			statlist_remove(&cancel_server->cancel_clients, &client->cancel_head);
```

Comment:
I am a bit unsure about the cancel_server part. Could we chat Monday to discuss this?
I'm not sure how cancel requests are queued, and new server connections given. If that is the case, I'm also now now sure where cancel_server is getting cancel requests queued, since it seems like the *new* server connection given to a cancel request has these cancel clients queued?

Reply:
I added some more comments to hopefully make this logic clearer.